src/lockscreen.js

```diff
@@ -86,12 +110,34 @@ var Lockscreen = GObject.registerClass({
         if (!this._key)
             return;
         if (this._manager.hasKey(this._key)) {
+            this._style.remove_class('locked');
+        }
+    }
 
-            /* TODO update the lock-screen graphics with can-be-opened image */
+    _getAssetsDir () {
+        if ('_lock' in this) {
+            var dir = Gio.File.new_for_path('/app/share/lockscreens/' + this._lock);
+            if (dir.query_exists(null))
+                return this._lock;
         }
+        return 'default';
     }
 
     _updateLockStateWithLock() {
+        var dir = this._getAssetsDir();
+
+        try {
+            this._css_provider.load_from_data(`
+widget.locked {
+    background-color: red;
+    background: url('file:///app/share/lockscreens/${dir}/locked.png') no-repeat center;
+}`
+                );
+        } catch (e) {
+            logError(e, 'Failed to load css');
+        }
+        this._playbin.uri = `file:///app/share/lockscreens/${dir}/animation.webm`;
+
         if (!this._lock)
             return;
         if (!this._manager.isUnlocked(this._lock))
```

Comment:
What needs to happen here is that:

a)  if the lock-screen `_lock` says its "locked" but the GSS says is now unlocked, then it means that the lock _just_ got unlock, therefore, the animation should be played.
 or

b) if the lock-screen `_lock` says its already unlock, and the GSS agrees, then it means it was already unlock, therefore we shouldn't even dislay the lock-screen anymore, neither the animation.